### PR TITLE
Support deb files as sources

### PR DESCRIPTION
**Pull request by sergiusens**
LP: #1604669

Signed-off-by: Sergio Schvezov <sergio.schvezov@ubuntu.com>

### CODE REVIEW

**Comment by elopio** [comment]
For an integration test, I think the right thing to do is to include a really small and simple deb in the repository, and use it as a source to stage a real snap.

**Comment by sergiusens** [comment]
@elopio done

**Comment by sergiusens** [comment]
retest this please

File: integration_tests/snaps/simple-deb/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  This integration tests verifies that deb files can be used as source
+  entries for a part.
+
+grade: devel
+confinement: devmode
+
+parts:
+  deb:
+    plugin: dump
+    source: small_0.1-1.deb
```
Comment (elopio) [inline integration_tests/snaps/simple-deb/snapcraft.yaml]: :100:

File: snapcraft/internal/sources.py [hunk 2]
```diff
@@ -58,6 +58,7 @@
 
 """
 
+import apt_inst
```
Comment (sergiusens) [inline snapcraft/internal/sources.py]: Fixed

File: snapcraft/internal/sources.py [hunk 3]
```diff
@@ -346,6 +346,36 @@ def provision(self, dst, clean_target=True, keep_zip=False):
...
+
+        if clean_target:
+            tmp_deb = tempfile.NamedTemporaryFile().name
+            shutil.move(deb_file, tmp_deb)
+            shutil.rmtree(dst)
+            os.makedirs(dst)
+            shutil.move(tmp_deb, deb_file)
+
+        deb = apt_inst.DebFile(deb_file)
+        print(deb, deb.data)
```
Comment (sergiusens) [inline snapcraft/internal/sources.py]: Done

File: snapcraft/tests/test_sources.py [hunk 4]
```diff
@@ -119,6 +119,55 @@ def test_extract_and_keep_zipfile(self, mock_zip):
...
 
 
+class TestDeb(tests.TestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.useFixture(fixtures.EnvironmentVariable(
+            'no_proxy', 'localhost,127.0.0.1'))
+        self.server = http.server.HTTPServer(
+            ('127.0.0.1', 0), FakeTarballHTTPRequestHandler)
```
Comment (sergiusens) [inline snapcraft/tests/test_sources.py]: Renamed to FakeFile...

**Comment by elopio** [comment]
retest this please

**Pull request merged**